server/src/test/java/io/crate/planner/WhereClauseOptimizerTest.java

```diff
@@ -215,4 +216,28 @@ public void test_filter_on_id_with_implicit_cast() {
         query = optimize("delete from t_pk where _id = 10 OR _id = true");
         assertThat(query.docKeys()).hasToString("Optional[DocKeys{'10'; 't'}]");
     }
+
+    // tracks a bug: https://github.com/crate/crate/issues/15395
+    @Test
+    public void test_filter_on_pk_with_or_true() {
+        WhereClauseOptimizer.DetailedQuery query = optimize("select * from t_pk where a = 1 or true");
```

Comment:
This doesn't relate to the bug fix, right?
Could you please double check that we don't test this functionality elsewhere?

Reply:
Thank you,

porting these tests from `AndOperatorTest` to `OrOperatorTest` then removed the ones with `or true`/`and false`:
```
    @Test
    public void testNormalizeBooleanTrueAndNonLiteral() throws Exception {
        assertNormalize("is_awesome and true", isReference("is_awesome"));
    }

    @Test
    public void testNormalizeBooleanFalseAndNonLiteral() throws Exception {
        assertNormalize("is_awesome and false", isLiteral(false));
    }
```